Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -389,17 +391,19 @@ inline void SeedFinder<external_spacepoint_t, platform_t>::filterCandidates(
     sorted_tops[i] = i;
   }
 
-  std::sort(sorted_bottoms.begin(), sorted_bottoms.end(),
-            [&state](const std::size_t& a, const std::size_t& b) -> bool {
-              return state.linCircleBottom[a].cotTheta <
-                     state.linCircleBottom[b].cotTheta;
-            });
-
-  std::sort(sorted_tops.begin(), sorted_tops.end(),
-            [&state](const std::size_t& a, const std::size_t& b) -> bool {
-              return state.linCircleTop[a].cotTheta <
-                     state.linCircleTop[b].cotTheta;
-            });
+  if (m_config.skipPreviousTopSP) {
+    std::sort(sorted_bottoms.begin(), sorted_bottoms.end(),
+              [&state](const std::size_t& a, const std::size_t& b) -> bool {
+                return state.linCircleBottom[a].cotTheta <
+                       state.linCircleBottom[b].cotTheta;
+              });
+
+    std::sort(sorted_tops.begin(), sorted_tops.end(),
+              [&state](const std::size_t& a, const std::size_t& b) -> bool {
+                return state.linCircleTop[a].cotTheta <
+                       state.linCircleTop[b].cotTheta;
+              });
+  }
```

Comment:
What is `skipPreviousTopSP` doing in the rest of the seeding code? Is it used to break a loop once we reach a specific condition?

Is sorting + breaking slower then not sorting at all and run on all the tops and bottoms?

Reply:
`skipPreviousTopSP + sorting` is used to avoid unnecessary iterations and speeds the code for PPP. But for SSS the sorting is lost during the filter because of the strip coordinates corrections that we apply, and if we break based on the previous sorting assumption we decrease the efficiency for SSS. So for SSS we have `skipPreviousTopSP==false` and we don't need to sort them.

Reply:
I understand. In other words, this will benefit only SSS

Reply:
Hi, just a procedural detail: leaving conversions open will block the `automerge` label, so if you want to get it merged, you need to close/resolve the conversations.

Reply:
I'm afraid I can't close the discussion despite being me the one who opened it... 

Anyway, I've given more thoughts on the `skipPreviousTopSP` and I believe that having a flag that a user can freely set  may be a huge problem. Especially if the effect is a drop of efficiency for SSS resulting from a break in the code which leads to wrong results. That smells more like a bug.

We can't rely on the user to know the seeding code that well imho. 

Maybe we should think about a solution that would make the ACTS code to figure this out by itself and properly apply the `skipPreviousTopSP` strategy or not. Something like "oh, there is a coordinate correction, so I cannot break the loop"

Reply:
`skipPreviousTopSP` could be replaced by `useDetailedDoubleMeasurementInfo` since they are related and we remove `skipPreviousTopSP` from the config. That is what I was planning to do with #2004

Reply:
should I do that in this PR?